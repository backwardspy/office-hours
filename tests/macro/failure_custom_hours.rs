use office_hours::office_hours;

fn main() {
    office_hours!(10, Clock::FourPm, {
        println!("I love my macro hours from 10:00 to 16:00")
    });
}