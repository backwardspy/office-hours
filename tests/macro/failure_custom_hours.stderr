error[E0308]: mismatched types
 --> tests/macro/failure_custom_hours.rs:4:19
  |
4 |       office_hours!(10, Clock::FourPm, {
  |       -             ^^ expected `Clock`, found integer
  |  _____|
  | |
5 | |         println!("I love my macro hours from 10:00 to 16:00")
6 | |     });
  | |______- arguments to this function are incorrect
  |
note: associated function defined here
 --> src/lib.rs
  |
  |     pub fn new(start: Clock, finish: Clock) -> Self {
  |            ^^^
